readme.md

```diff
@@ -1 +1,73 @@
-http://todomvc.com but for back end.
+# Goal
+
+* Create various implementations of a minimally simple addressbook service
+
+# Why?
+
+* To ease comparison between different languages/frameworks/approaches
+* To demonstrate how to use the various languages/frameworks/approaches
+
+## The Original Idea
+
+http://todomvc.com but using server-side technologies.
+
+* have a database with a table with names and phone numbers (and probably a surrogate key)
+* the application should list all the entries (optionally paged)
+* you should be able to add, edit and delete an entry
+* each entry has 2 fields: name and phone number
+* *any* other features are optional
+
+
+# Implementation Guidelines
+
+## General
+
+* Go for clarity and simplicity which should ease comparison and improve the pedagogical win.
+* Err on the side of excessive commenting.
+* Implementations MUST include persistence.
+* Implementations SHOULD behave as described in the API doc below, *whether it provides an actual API service or merely outputs HTML directly*.
+
+## Contacts
+
+A contact is completely described as:
+
+```json
+{
+  "id":   <integer>,
+  "name": "<string>",
+  "phone": "<string>"
+}
+```
+
+### SQL
+
+If you choose to use an SQL-based persistence, here is some example [DDL]:
+
+```sql
+CREATE TABLE contacts (
+id INTEGER NOT NULL PRIMARY UNIQUE AUTOINCREMENT,
+name TEXT NOT NULL,
+phone TEXT NOT NULL
+);
+```
```

Comment:
Hmm, my table is called "person" rather than "contacts": https://github.com/pdurbin/addressbookmvc/blob/890dbdb6a96a10430350dcf533a605a4ad317bf9/examples/javaee7/src/main/java/com/greptilian/addressbookmvc/javaee7/Person.java

And my fields are "displayname" and "phonenumber".

The table and fields in the implementation by @aditsu are different still. The table is "phonebook" and he uses "phone_number":  https://github.com/pdurbin/addressbookmvc/blob/db830828f668767f0c9afeeb33cdee3cab4e0f05/examples/wicket/setup.sql

If we're going to specify the tables, should they all be the same? Or is this truly just an example? Could people use UUIDs rather than integers for "id" or would that be non-compliant.